ravenframework/Runners/DistributedMemoryRunner.py

```diff
@@ -66,25 +72,26 @@ def isDone(self):
     if not self.started:
       return False
 
-    if self.thread is None:
-      return True
-    else:
-      if im.isLibAvail("ray"):
-        try:
-          ray.get(self.thread, timeout=waitTimeOut)
-          return True
-        except ray.exceptions.GetTimeoutError:
-          return False
-        except ray.exceptions.RayTaskError:
-          #The code gets this undocumented error, and
-          # I assume it means the task has unfixably died,
-          # and so is done
-          return True
-        #Alternative that was tried:
-        #return self.thread in ray.wait([self.thread], timeout=waitTimeOut)[0]
-        #which ran slower in ray 1.9
+    with self.__funcLock:
+      if self.__func is None:
+        return True
       else:
-        return self.thread.finished
+        if im.isLibAvail("ray"):
+          try:
+            ray.get(self.__func, timeout=waitTimeOut)
+            return True
+          except ray.exceptions.GetTimeoutError:
+            return False
+          except ray.exceptions.RayTaskError:
+            #The code gets this undocumented error, and
+            # I assume it means the task has unfixably died,
+            # and so is done
+            return True
```

Comment:
`True` means the job probably ended successfully? Is that the right return here? Do we believe the job probably finished, but the task didn't terminate as expected?

Reply:
Hm, well, isDone can either return true or false, and if it returns false, then JobHandler will think that it is still running, and so not remove it.  Setting returnCode = -1 might be the correct thing to do here.

Reply:
Oh, okay, I misread, this is just telling if the guy is done nor not. `True` seems like a fair response.

Reply:
(actually, this might be the cause of #1901 )

Reply:
FYI: For future reference, the reason that returnCode = -1 needs to be set here as opposed to _collectRunnerResponse (which also uses ray.get) is that returnCode is checked before _collectRunnerResponse is called.